File: hoosong0235/Silver/4673. 셀프 넘버/셀프 넘버.py
def d(n):
    sum = 0
    for i in str(n):
        sum += int(i)
    return n + sum


list = []
for i in range(10000):
    list.append(1)

while (1 in list):
    index = list.index(1)
    value = index + 1
    list[index] = True
    while (value <= 10000):
        list[value - 1] = False
        value = d(value)
    print(index + 1)

File: hoosong0235c/Silver/4673. 셀프 넘버/셀프 넘버.c
#include <stdio.h>

int d(int n);

int main(void) {
    int a[10000] = { [0 ... 9999] = 1 };
    for (int i = 0; i < 10000; i++) if (d(i) <= 10000) a[d(i) - 1] = 0;
    for (int i = 0; i < 10000; i++) if (a[i]) printf("%d\n", i + 1);

    return 0;
}

int d(int n) {
    int sum = n;
    for (int i = 1; n >= i; i *= 10) sum += (n % (i * 10) - n % i) / i;
    return sum;
}